chain/xrp/builder/builder.go

```diff
@@ -0,0 +1,93 @@
+package builder
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	xc "github.com/cordialsys/crosschain"
+	"github.com/cordialsys/crosschain/chain/xrp/tx"
+	xrptxinput "github.com/cordialsys/crosschain/chain/xrp/tx_input"
+	binarycodec "github.com/xyield/xrpl-go/binary-codec"
+)
+
+// TxBuilder for Template
+type TxBuilder struct {
+	Asset xc.ITask
+}
+
+var _ xc.TxBuilder = &TxBuilder{}
+
+type TxInput = xrptxinput.TxInput
+
+// NewTxBuilder creates a new Template TxBuilder
+func NewTxBuilder(cfgI xc.ITask) (xc.TxBuilder, error) {
+	return TxBuilder{
+		Asset: cfgI,
+	}, nil
+}
+
+// NewTransfer creates a new transfer for an Asset, either native or token
+func (txBuilder TxBuilder) NewTransfer(from xc.Address, to xc.Address, amount xc.AmountBlockchain, input xc.TxInput) (xc.Tx, error) {
+	if _, ok := txBuilder.Asset.(*xc.TokenAssetConfig); ok {
+		return txBuilder.NewTokenTransfer(from, to, amount, input)
+	}
+	return txBuilder.NewNativeTransfer(from, to, amount, input)
+}
+
+// NewNativeTransfer creates a new transfer for a native asset
+func (txBuilder TxBuilder) NewNativeTransfer(from xc.Address, to xc.Address, amount xc.AmountBlockchain, input xc.TxInput) (xc.Tx, error) {
+	txInput := input.(*TxInput)
+
+	signingPubKeyHex := hex.EncodeToString(txInput.Pubkey)
+	//txInput.XRPTx.SigningPubKey = signingPubKeyHex
+	fmt.Println(txInput)
+
+	jsonData, err := json.Marshal(txInput.XRPTx)
```

Comment:
You're only partially filling some fields in `txInput.XRPTx`, so it's hard to follow what is actually getting serialized/copied out.  It'd be more clear to explicitly define what our inputs are and then copy them here (I think just `LastLedgerSequence`, and `Sequence`)